### PR TITLE
[plugin.video.manoto] 2.3.0

### PR DESCRIPTION
**Pull request by bbk79**
### Description
Updated for matrix / python3

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.manoto/addon.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+       version="2.3.0"
+       provider-name="babak">
+  <requires>
+    <import addon="xbmc.python" version="3.0.0"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource" library="default.py">
+        <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Manoto TV</summary>
```
Comment (enen92) [inline plugin.video.manoto/addon.xml]:
```suggestion
    <summary lang="en_GB">Manoto TV</summary>
```

File: plugin.video.manoto/addon.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <description lang="en">Manoto TV is a specialist channel broadcasting in Farsi language targeting viewers in Europe and The Middle East. It is a general entertainment channel and comprises a mix of television programmes acquired from other broadcasters and programmes produced by the Marjan TV team at Wimbledon Studios.</description>
```
Comment (enen92) [inline plugin.video.manoto/addon.xml]:
```suggestion
    <description lang="en_GB">Manoto TV is a specialist channel broadcasting in Farsi language targeting viewers in Europe and The Middle East. It is a general entertainment channel and comprises a mix of television programmes acquired from other broadcasters and programmes produced by the Marjan TV team at Wimbledon Studios.</description>
```

File: plugin.video.manoto/addon.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <platform>all</platform>
+    <language>fa</language>
+    <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline plugin.video.manoto/addon.xml]:
```suggestion
    <license>GPL-2.0-only</license>
```

or 

```suggestion
    <license>GPL-2.0-or-later</license>
```

File: plugin.video.manoto/default.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import xbmc, xbmcgui, os, xbmcaddon, xbmcplugin, urllib.request, urllib.error, urllib.parse, re
+
+def main():
+    __settings__ = xbmcaddon.Addon()
+    home = __settings__.getAddonInfo('path')
+    resources = os.path.join(home, 'resources')
+    addon_handle = int(sys.argv[1])
+
+    icons = {}
+    icons['Auto'] = xbmc.translatePath(os.path.join(resources, 'icon.png'))
```
Comment (enen92) [inline plugin.video.manoto/default.py]:
```suggestion
    icons['Auto'] = xbmcvfs.translatePath(os.path.join(resources, 'icon.png'))
```

File: plugin.video.manoto/default.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    icons['720'] = xbmc.translatePath(os.path.join(resources, 'hd.png'))
```
Comment (enen92) [inline plugin.video.manoto/default.py]:
```suggestion
    icons['720'] = xbmcvfs.translatePath(os.path.join(resources, 'hd.png'))
```

File: plugin.video.manoto/default.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    icons['1080'] = xbmc.translatePath(os.path.join(resources, 'fhd.png'))
```
Comment (enen92) [inline plugin.video.manoto/default.py]:
```suggestion
    icons['1080'] = xbmcvfs.translatePath(os.path.join(resources, 'fhd.png'))
```

File: plugin.video.manoto/default.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+         try:
+              li.setArt({'thumb': icons[streams[bitrate][2]]})
+         except IndexError:
+              li.setArt({'thumb': icon})
+         xbmcplugin.addDirectoryItem(handle=addon_handle, url=streams[bitrate][1], listitem=li, isFolder=False)
+
+    xbmcplugin.endOfDirectory(addon_handle)
+
+def getStreamsFromPlayList():
+    base_url = "https://d2rwmwucnr0d10.cloudfront.net/"
```
Comment (enen92) [inline plugin.video.manoto/default.py:32]: this link is a bit suspect. Who is in charge of this, where does the link come from?

**Review by enen92 — CHANGES_REQUESTED** [review]

File: plugin.video.manoto/default.py (hunk 7 above)
Comment (bbk79) [reply plugin.video.manoto/default.py:32]: it's from https://www.manototv.com/live, this plugin is known to them and it was suggested to use the hard-coded URL in a previous conversation with @keithah and @narimangharib.

**Comment by bbk79** [comment]
Did I somehow break Travis? :)

**Comment by enen92** [comment]
Nop it just broke itself :). 
We've basically hit the monthly quota limit...which means it's time to move away from travis.
I'll have a look on Friday, unfortunately until then we can't do much

**Review by enen92 — APPROVED** [review]

**Pull request merged**